src/main/java/uk/gov/hmcts/darts/cases/mapper/AdvancedSearchRequestMapper.java

```diff
@@ -0,0 +1,151 @@
+package uk.gov.hmcts.darts.cases.mapper;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import uk.gov.hmcts.darts.cases.model.GetCasesSearchRequest;
+import uk.gov.hmcts.darts.cases.model.SQLQueryModel;
+import uk.gov.hmcts.darts.common.entity.CaseEntity;
+import uk.gov.hmcts.darts.common.entity.CourtroomEntity;
+import uk.gov.hmcts.darts.common.entity.HearingEntity;
+
+@Slf4j
+public class AdvancedSearchRequestMapper {
+
+    private boolean joinHearing = false;
+    private boolean joinCourtroom = false;
+    private boolean joinCourthouse = false;
+    private boolean joinJudges = false;
+    private boolean joinDefendants = false;
+    private boolean joinEvents = false;
+
+    SQLQueryModel sqlModel = new SQLQueryModel();
+
+    public SQLQueryModel mapToSQL(GetCasesSearchRequest request) {
```

Comment:
This method could benefit from applying some of the functional programming principles outlined in this page https://www.baeldung.com/java-functional-programming 

Specifically, refer to the section about side affects: "Side effects can be anything apart from the intended behaviour of the method. For instance, side effects can be as simple as updating a local or global state". 
At present, it looks like this method is at risk of side effects, if for example another method is introduced in the future which updates the value of those same class variables. You could end up with unintended side effects, with this method potentially producing different results, depending on the order in which the method is called against an object.